### ISSUE TITLE
Request: A quantile function for ordinal data only

### ISSUE THREAD

**Issue by pdeffebach** [JuliaLang/julia#27367]
There are a variety of types of data for which order is defined, but not other mathematical operations. It seems to be the consensus, for instance, that the `Date` type should not have `+` or `/` defined for it. 

However, if you have a vector of dates, you might still want to know the "quantiles" of those dates. If you can sort a vector of dates, because `<` is defined, you can ask "What is the 25th percentile of dates in my vector?" 

You can't do this with the current `quantile` function, because in the case of a tie, it finds a midpoint between the two values by taking a mean. 

`R`'s quantile function has the keyword argument `Type`, and when you call `quantile(x, ..., Type = 1)` it returns the lower of the two values in the case of a tie. 

I am currently working on a better `describe` function for returning summary statistics of a `DataFrame`, and think it would be useful to return a quantile-like value for ordinal data. Unfortunately, such a function is not defined either here or in `StatsBase`. 

`quantile` is a super well-written function in `Base`, being clever enough to only sort values between the minimum and maximum percentiles asked for. Writing an `ordinal quantile` function in `StatsBase` would essentially mean re-writing the `quantile` function entirely. Rather, I think it makes sense to add a new method, call it `ordquantile` or something that keeps everything in the current `quantile` function *except* for the part that takes the mean of ties, and returns the lower value instead. 

Does this reasoning make sense for it to live here?

**Comment by nalimilan** [comment]
How about adding an argument to `quantile`?

**Comment by pdeffebach** [comment]
For reference [here](https://github.com/JuliaLang/julia/blob/8e6cfff89c4b936e797cd2d7899b7a31c7dccbd3/base/statistics.jl#L249) is the relevant code. We would just add a new function that's just like `@inline function _quantile(v::AbstractVector, p::Real)` that gets called with a keyword argument `ordinal = true`.

**Comment by andreasnoack** [comment]
I agree that this would be useful. This was discussed a bit in https://github.com/JuliaLang/julia/issues/19190#issuecomment-257885325, https://github.com/JuliaLang/julia/pull/19359#issuecomment-261580952, and in the very old https://github.com/JuliaLang/julia/issues/1333.

**Comment by matthieugomez** [comment]
Would it be so bad to change the default quantile to type 1? It sounds that it would simplify a lot things. Stata does it by default.

**Comment by matthieugomez** [comment]
Yes you are right. Stata uses type 2 which does give the median as the average of the two central elements.

**Comment by pdeffebach** [comment]
To be clear, julia doesn't have implementations for all the different quantile versions described [here](https://www.rdocumentation.org/packages/stats/versions/3.5.0/topics/quantile), right? 

Should there be an effort to implement all 7? seems excessive, but maybe there is more demand.

**Comment by nalimilan** [comment]
No, we only support one variant currently. I'm not sure there's a real demand to support all variants, most software only implement a few of them. But having a way to compute them would still be useful to replicate results, either in a package or in Base.

**Comment by nalimilan** [comment]
This will be fixed by https://github.com/JuliaStats/Statistics.jl/pull/187.
